File: src/main/scala/com/xm4399/util/OtherUtil.java
package com.xm4399.util;

import java.io.IOException;
import java.io.PrintWriter;
import java.io.StringWriter;
import java.io.Writer;

/**
 * @Auther: czk
 * @Date: 2020/9/1
 * @Description:
 */
public class OtherUtil {
    // 将异常信息转为string
    public  String getException(Exception e) {
        Writer writer = null;
        PrintWriter printWriter = null;
        try {
            writer = new StringWriter();
            printWriter = new PrintWriter(writer);
            e.printStackTrace(printWriter);
            return writer.toString();
        } finally {
            try {
                if (writer != null)
                    writer.close();
                if (printWriter != null)
                    printWriter.close();
            } catch (IOException e1) {
                e1.printStackTrace();
            }
        }
    }
}

File: src/main/scala/com/xm4399/util/OtherUtil.scala
package com.xm4399.util

/**
 * @Auther: czk
 * @Date: 2020/9/1 
 * @Description:
 */
object OtherUtil {
  // 判断两个map是否相同
  def isSameFromTwoMap (map1: Map[String,Any], map2: Map[String,Any]) : Boolean={
    if (map1.size != map2.size){
      return false
    }
    for (key <- map1.keys){
      if (!map2.contains(key)){
        return  false
      } else {
        if (!map2.get(key).equals(map1.get(key))){
          return  false
        }
      }
    }
    true
  }
}
